tests/main/snap-quota-services/task.yaml

```diff
@@ -0,0 +1,76 @@
+summary: Basic test for quota-related snap commands.
+
+details: |
+  Basic test for snap quota, remove-quota and quotas commands.
+
+# In arm devices using ubuntu core, memory quota cannot be used because
+# memory cgroup is disabled
+systems: [ -ubuntu-core-*-arm-* ]
+
+prepare: |
+  snap install test-snapd-stressd --edge --devmode
+  tests.cleanup defer snap remove --purge test-snapd-stressd
+  snap set system experimental.quota-groups=true
+  tests.cleanup defer snap unset system experimental.quota-groups
+
+execute: |
+  if os.query is-trusty || os.query is-amazon-linux || os.query is-centos 7 || os.query is-xenial || os.query is-core16; then
+    # just check that we can't do anything with quota groups on systems with
+    # old systemd versions, we need at least 230 to avoid buggy slice usage
+    # reporting
+
+    snap set-quota foobar --memory=1MB 2>&1 | tr '\n' ' ' | tr -s ' ' | MATCH "systemd version 2[0-2][0-9] is too old \(expected at least 230\)"
+    exit 0
+  fi
+
+  # Create the top level snap group with a memory limit. test-snapd-stressd
+  # contains 4 services, which makes it perfect for our testing as we only need
+  # this one snap to test isolating services.
+  snap set-quota test-top --memory=400MB test-snapd-stressd
+
+  echo "Create a sub-level group for the stress service and put it into that"
+  snap set-quota test-sub --parent=test-top --cpu=10% test-snapd-stressd.stress-sc
+
+  echo "Test the quota groups listings look sane"
+  snap quotas | cat -n > quotas.txt
+  MATCH "     1\s+Quota\s+Parent\s+Constraints\s+Current$" < quotas.txt
+  MATCH "     2\s+test-top\s+memory=400MB(\s*|\s*memory=[0-9.a-zA-Z]+)+$" < quotas.txt
+  MATCH "     3\s+test-sub\s+test-top\s+cpu=10%\s+$" < quotas.txt
+
+  echo "Test that service details are listed correctly in test-sub"
+  snap quota test-sub | cat -n > details.txt
+  MATCH "     1\s+name:\s+test-sub$" < details.txt
+  MATCH "     2\s+parent:\s+test-top$" < details.txt
+  MATCH "     3\s+constraints:$" < details.txt
+  MATCH "     4\s+cpu-count:\s+0$" < details.txt
+  MATCH "     5\s+cpu-percentage:\s+10$" < details.txt
+  MATCH "     6\s+current:$" < details.txt
+  MATCH "     7\s+services:$" < details.txt
+  MATCH "     8\s+-\s+test-snapd-stressd.stress-sc$" < details.txt
+
+  echo "Test that it fails when we apply a journal quota to the sub-group"
+  snap set-quota test-sub --journal-size=128MB 2>&1 | tr '\n' ' ' | tr -s ' ' | MATCH 'error: cannot update quota group: cannot update group "test-sub": journal quotas are not supported for individual services'
+
+  echo "Test that we can however put a journal quota on the parent group"
+  snap set-quota test-top --journal-size=128MB
+
+  ls /nono
```

Comment:
Is this a debug leftover?

Reply:
Yes! It's a debug leftover, I'll remove this later today 😃

Reply:
Fixed :)